File: esports-backend/src/main/java/com/esports/entity/UserProfile.java
package com.esports.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.validator.constraints.URL;
import java.time.LocalDateTime;

@Entity
@Table(name = "user_profiles")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @OneToOne
    @JoinColumn(name = "user_id", unique = true)
    private User user;
    
    @Pattern(regexp = "^(\\S+\\s+){0,29}\\S*$", message = "Bio must not exceed 30 words")
    private String bio;
    
    private String location;
    
    @URL(message = "Website must be a valid URL")
    private String website;
    
    @Pattern(regexp = "^(\\S+\\s+){0,29}\\S*$", message = "Skills must not exceed 30 words")
    private String skills;
    
    @Column(name = "professional_name")
    private String professionalName;
    
    @Column(name = "channel_name")
    private String channelName;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    public Long getUserId() { 
        return user != null ? user.getId() : null; 
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: esports-backend-dotnet/Upkart1/Entities/UserProfile.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Upkart1.Entities
{
    [Table("user_profiles")]
    public class UserProfile
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [ForeignKey(nameof(User))]
        [Column("user_id")]
        public long UserId { get; set; }

        [JsonIgnore]
        public User User { get; set; } = null!;

        [RegularExpression(@"^(\S+\s+){0,29}\S*$", ErrorMessage = "Bio must not exceed 30 words")]
        public string? Bio { get; set; }

        public string? Location { get; set; }

        [Url(ErrorMessage = "Website must be a valid URL")]
        public string? Website { get; set; }

        [RegularExpression(@"^(\S+\s+){0,29}\S*$", ErrorMessage = "Skills must not exceed 30 words")]
        public string? Skills { get; set; }

        [Column("professional_name")]
        public string? ProfessionalName { get; set; }

        [Column("channel_name")]
        public string? ChannelName { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
